src/classes/linked-list/linked-list.spec.ts

```diff
@@ -0,0 +1,388 @@
+        beforeEach(() => {
+          isElementInserted = linkedList.insertAt(elementToInsert, initialSizeOfList + 1);
+        });
+
+        it('Should not add element', () => {
+          expect(isElementInserted).to.be.false;
+          expect(linkedList.length).to.be.equal(initialSizeOfList);
+        });
+      });
+
+      describe('When insert a element in first index', () => {
+        beforeEach(() => {
+          isElementInserted = linkedList.insertAt(elementToInsert, 0);
+        });
+
+        it('Should add the element in first position', () => {
+          expect(isElementInserted).to.be.true;
+          expect(linkedList.length).to.be.equal(initialSizeOfList + 1);
+          expect(linkedList.getElementAt(0)).to.be.equal(elementToInsert);
+        });
+      });
+
+      describe('When insert a element in last position', () => {
+        beforeEach(() => {
+          isElementInserted = linkedList.insertAt(elementToInsert, initialSizeOfList);
+        });
+
+        it('Should add the element in last position', () => {
+          expect(isElementInserted).to.be.true;
+          expect(linkedList.length).to.be.equal(initialSizeOfList + 1);
+          expect(linkedList.getElementAt(initialSizeOfList)).to.be.equal(elementToInsert);
+        });
+      });
+
+      describe('When insert a element in another valid index', () => {
+        const indexToInsert = initialSizeOfList - 1;
+
+        beforeEach(() => {
+          isElementInserted = linkedList.insertAt(elementToInsert, indexToInsert);
+        });
+
+        it('Should add the element in the specified index', () => {
+          expect(isElementInserted).to.be.true;
+          expect(linkedList.length).to.be.equal(initialSizeOfList + 1);
+          expect(linkedList.getElementAt(indexToInsert)).to.be.equal(elementToInsert);
+        });
+      });
+    });
+  });
+
+  describe('removeAt', () => {
+    let removedElement: any;
+
+    describe('Given an empty linkedList', () => {
+      beforeEach(() => {
+        linkedList = new LinkedList();
+        expect(linkedList.length).to.be.equal(0);
+      });
+
+      describe('When try to remove an element in any position', () => {
+        beforeEach(() => {
+          removedElement = linkedList.removeAt(-1);
+        });
+
+        it('Should return undefined', () => {
+          expect(linkedList.removeAt(-1)).to.be.undefined;
+          expect(linkedList.removeAt(0)).to.be.undefined;
+          expect(linkedList.removeAt(1)).to.be.undefined;
+        });
+      });
+    });
+
+    describe('Given a linkedList with 3 elements', () => {
+      const originalElements = [1, 2, 3, 4, 5];
+
+      beforeEach(() => {
+        linkedList = new LinkedList();
+        originalElements.forEach(element => linkedList.push(element));
+        expect(linkedList.length).to.be.equal(originalElements.length);
+      });
+
+      describe('When try to remove an element in an index equal the size of list', () => {
```

Comment:
When try to remove an element in an index equal the size of list --> When trying to remove an element at an index equal to the length of the list